tests/dataTransformer.spec.ts

```diff
@@ -0,0 +1,198 @@
+    encoded.typecode.should.equal(117);
+    isBuffer(encoded.content).should.equal(true);
+    const decoded: any = transformer.decode(encoded);
+    decoded.should.equal(booleanBody);
+    done();
+  });
+
+  it("should correctly encode/decode a null message body", function (done) {
+    const encoded: any = transformer.encode(nullBody);
+    encoded.typecode.should.equal(117);
+    isBuffer(encoded.content).should.equal(true);
+    const decoded: any = transformer.decode(encoded);
+    should.equal(decoded, nullBody);
+    done();
+  });
+
+  it("should correctly encode/decode an undefined message body", function (done) {
+    const encoded: any = transformer.encode(undefinedBody);
+    encoded.typecode.should.equal(117);
+    isBuffer(encoded.content).should.equal(true);
+    const decoded: any = transformer.decode(encoded);
+    decoded.should.equal("");
+    done();
+  });
+
+  it("should correctly encode/decode an empty string message body", function (done) {
+    const encoded: any = transformer.encode(emptyStringBody);
+    encoded.typecode.should.equal(117);
+    isBuffer(encoded.content).should.equal(true);
+    const decoded: any = transformer.decode(encoded);
+    decoded.should.equal(emptyStringBody);
+    done();
+  });
+
+  it("should correctly encode/decode an array message body", function (done) {
+    const encoded: any = transformer.encode(arrayBody);
+    encoded.typecode.should.equal(117);
+    isBuffer(encoded.content).should.equal(true);
+    const decoded: any = transformer.decode(encoded);
+    assert.deepEqual(decoded, arrayBody);
+    done();
+  });
+
+  it("should correctly encode/decode an object message body", function (done) {
+    const encoded: any = transformer.encode(objectBody);
+    encoded.typecode.should.equal(117);
+    isBuffer(encoded.content).should.equal(true);
+    const decoded: any = transformer.decode(encoded);
+    assert.deepEqual(decoded, objectBody);
+    done();
+  });
+
+  it("should correctly encode/decode a buffer message body", function (done) {
+    const encoded: any = transformer.encode(bufferbody);
+    encoded.typecode.should.equal(117);
+    isBuffer(encoded.content).should.equal(true);
+    const decoded: any = transformer.decode(encoded);
+    assert.deepEqual(decoded, bufferbody);
+    done();
+  });
+
+  it("should correctly encode/decode a hex buffer message body", function (done) {
+    const encoded: any = transformer.encode(hexBufferBody);
+    encoded.typecode.should.equal(117);
+    isBuffer(encoded.content).should.equal(true);
+    const decoded: any = transformer.decode(encoded);
+    assert.deepEqual(decoded, hexBufferBody);
+    done();
+  });
+
+  describe("decode", function () {
+    // It is possible that we receive an AMQP value type from the messages that were sent with
+    // previously shipped version of the sdk. If so then we should be able to handle those scenarios.
+    it("should correctly decode a string message body", function (done) {
```

Comment:
We will send the received message body as-is if we cannot decode. This will ensure previous behavior of receiving AMQP value types remains unchanged.